src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            using (JsonDocument employeesToSend = JsonDocument.Parse(jsonString))
+            {
+                // regular send:
+                Mailbox.SendAllEmployeesData(employeesToSend.RootElement);
+
+                // modified elements send:
+                JsonNode modifiableDocument = JsonElement.DeepCopy(employeesToSend);
+                var employees = modifiableDocument as JsonObject;
+                Assert.Equal(2, employees.Count());
+
+                employees.Add(EmployeesDatabase.GetNextEmployee());
+                Assert.Equal(3, employees.Count());
+
+                Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+            }
+        }
+
+        /// <summary>
+        /// Parsing right to JsonNode if user knows data will be modified
+        /// </summary>
+        [Fact]
+        public static void TestParsingToJsonNode()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+            employees.Add(EmployeesDatabase.GetNextEmployee());
+            Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+
+            Assert.Equal(2, employees.Count());
+            Assert.True(employees.ContainsProperty("employee1"));
+            Assert.True(employees.ContainsProperty("employee2"));
+
+            JsonObject employee2 = employees.GetObjectProperty("employee2");
+            Assert.IsType<JsonString>(employee2["name"]);
+            Assert.Equal("Zoe", (JsonString)employee2["name"]);
+            Assert.IsType<JsonString>(employee2["surname"]);
+            Assert.Equal("Coder", (JsonString)employee2["surname"]);
+            Assert.IsType<JsonNumber>(employee2["age"]);
+            Assert.Equal(24, (JsonNumber)employee2["age"]);
+        }
+
+        /// <summary>
+        /// Copying JsoneNode
+        /// </summary>
+        [Fact]
+        public static void TestCopyingJsonNode()
+        {
+            JsonObject employee = EmployeesDatabase.GetManager();
+            JsonNode employeeCopy = JsonNode.DeepCopy(employee);
+
+            static bool recursiveEquals(JsonNode left, JsonNode right)
```

Comment:
Addressing question from previous PR: I wrote here recursive equals, because I needed to check if `DeepCopy` making a copy of `JsonNode` works well, but I'm not sure if users would need it (maybe also for some of their tests).